src/app/frontend/deploy/deployfromsettings_controller.js

```diff
@@ -183,6 +190,33 @@ export default class DeployFromSettingsController {
   }
 
   /**
+   * @returns {!angular.$q.Promise}
+   * @export
+   */
+  updateAvailableProtocols() {
```

Comment:
Hmmm... The more I think about this solution, the more problems I find. I.e., this code has race conditions when you change this.isExternal until this promise fulfills. Also, on changing this.isExternal, suddenly all your protocol selections get changed too. 

How about making this an async validator? Like the unique-name-validator? It can connect to angular form processing pipeline to take care of race conditions and would display message like: "this protocol is not supported for external services". The benefit of the error message is that user immediately sees what is wrong and what should/can be change. When we silently change protocols of all ports, this may be unnoticed.